scripts/quests/otherAreas/In_Search_of_the_Truth.lua

```diff
@@ -0,0 +1,315 @@
+
+            ['Raminey'] =
+            {
+                onTrigger = function(player, npc)
+                    if not quest:isVarBitsSet(player, 'Option', 1) then
+                        return quest:progressEvent(549)
+                    end
+                end,
+            },
+            ['Zadant'] =
+            {
+                onTrigger = function(player, npc)
+                    if not quest:isVarBitsSet(player, 'Option', 2) then
+                        return quest:progressEvent(551)
+                    end
+                end,
+            },
+            ['Fouagine'] =
+            {
+                onTrigger = function(player, npc)
+                    if not quest:isVarBitsSet(player, 'Option', 3) then
+                        return quest:progressEvent(553)
+                    end
+                end,
+            },
+            ['Noam'] =
+            {
+                onTrigger = function(player, npc)
+                    if not quest:isVarBitsSet(player, 'Option', 4) then
+                        return quest:progressEvent(555)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [549] = function(player, csid, option, npc)
+                    quest:setVarBit(player, 'Option', 1)
+                end,
+
+                [551] = function(player, csid, option, npc)
+                    quest:setVarBit(player, 'Option', 2)
+                end,
+
+                [553] = function(player, csid, option, npc)
+                    quest:setVarBit(player, 'Option', 3)
+                end,
+
+                [555] = function(player, csid, option, npc)
+                    quest:setVarBit(player, 'Option', 4)
+                end,
+
+                [557] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:setVar(player, 'Option', 0)
+                        quest:setVar(player, 'Prog', 1)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        -- Second portion of quest
+        -- Does not care if you talked to Mengrenaux or Chemioue. Will progress if you enter the correct order after talking to Tressia.
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED and
+            quest:getVar(player, 'Prog') == 1
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            ['Tressia'] =
+            {
+                onTrigger = function(player, npc)
+                        return quest:progressEvent(565, 0, 0)
+                end,
+            },
+
+            ['Mengrenaux'] = quest:progressEvent(560),
+
+            ['Chemioue'] = quest:progressEvent(561),
+
+            onEventFinish =
+            {
+                [560] = function(player, csid, option, npc)
+                end,
+
+                [561] = function(player, csid, option, npc)
+                end,
```

Comment:
Why are there empty onEventFinish?

Reply:
Delete those 2, since they do nothing, and also use quest:event() for events 560and 561